### DISCUSSION TITLE
What's the best way to debug a `cue.Value` in debug console?

### DISCUSSION THREAD

**Discussion by Rots** [cuelang/cue#812] [Q&A] [answered]
I'm trying some simple conversions and using VSCode debugger to step through my program, but the cue values are hard to reach in the debug mode.

Is there a best practise to quickly see the current state of the `cue.Value` (e.g. get the output in .cue format).
Calling `call v.String()` gives me
"", <error(*cuelang.org/go/cue.valueError)>)
even though I don't expect that.
`call v.Path().String()` is somewhat useful.

Any tips welcome.

**Comment by mpvl** [comment, marked as answer]
CUE values implement `fmt.Formatter`, so you can use `fmt.Sprint(v)` and friends. You can get different levels of details by using `fmt.Sprintf("%+v", v)` and `fmt.Sprintf("%#v", v)`.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/812.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
